source/MapPanel.cpp

```diff
@@ -1063,31 +1074,36 @@ void MapPanel::DrawWormholes()
 
 		for(auto &&link : it.second.Links())
 			if(!link.first->Inaccessible() && !link.second->Inaccessible() && p.IsInSystem(link.first)
-					&& player.HasVisited(*link.first) && player.HasVisited(*link.second))
-				arrowsToDraw.emplace(link.first, link.second);
+				&& player.HasVisited(*link.first) && player.HasVisited(*link.second))
+				arrowsToDraw.emplace_back(link.first, link.second, it.second.GetLinkColor());
+
 	}
 
-	const Color &wormholeDim = *GameData::Colors().Get("map unused wormhole");
-	const Color &arrowColor = *GameData::Colors().Get("map used wormhole");
 	static const double ARROW_LENGTH = 4.;
 	static const double ARROW_RATIO = .3;
 	static const Angle LEFT(30.);
 	static const Angle RIGHT(-30.);
 	const double zoom = Zoom();
 
-	for(const pair<const System *, const System *> &link : arrowsToDraw)
+	for(const WormholeArrow &link : arrowsToDraw)
 	{
+		// Get the wormhole link color.
+		const Color &arrowColor = *link.color;
+		const Color &wormholeDim = Color::Combine(1.f, arrowColor, -0.66f, arrowColor);
```

Comment:
@Azure3141 I've opened a PR to this branch that adds a `Multiply` method to `Color`.
[PR](https://github.com/Azure3141/endless-sky/pull/4)
With that, this line can become:
```suggestion
		const Color &wormholeDim = Color::Multiply(.33f, arrowColor);
```